apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-contentlets/dot-palette-contentlets.component.ts

```diff
@@ -0,0 +1,125 @@
+import { take } from 'rxjs/operators';
+import { DotPaletteInputFilterComponent } from '../dot-palette-input-filter/dot-palette-input-filter.component';
+
+@Component({
+    selector: 'dot-palette-contentlets',
+    templateUrl: './dot-palette-contentlets.component.html',
+    styleUrls: ['./dot-palette-contentlets.component.scss']
+})
+export class DotPaletteContentletsComponent implements OnChanges {
+    @ViewChild('filterInput', { static: true })
+    filterInput: DotPaletteInputFilterComponent;
+
+    @Input() contentTypeVariable: string;
+    @Output() hide = new EventEmitter();
+    items: DotCMSContentlet[] | DotCMSContentType[];
+    isFormContentType: boolean;
+    hideNoResults = true;
+
+    constructor(
+        public paginatorESService: DotPaginatorESContentService,
+        public paginationService: PaginatorService,
+        private dotContentletEditorService: DotContentletEditorService
+    ) {}
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.contentTypeVariable?.currentValue) {
+            this.isFormContentType = changes?.contentTypeVariable?.currentValue === 'forms';
+
+            if (this.isFormContentType) {
+                this.paginationService.url = `v1/contenttype`;
+                this.paginationService.paginationPerPage = 15;
+                this.paginationService.sortField = 'modDate';
+                this.paginationService.setExtraParams('type', 'Form');
+                this.paginationService.sortOrder = 1;
+            } else {
+                this.paginatorESService.setExtraParams('+contentType', this.contentTypeVariable);
+                this.paginatorESService.deleteExtraParams('+title');
+                this.paginatorESService.paginationPerPage = 15;
+            }
+
+            this.loadData();
+        }
+    }
+
+    /**
+     * Loads data through pagination service
+     *
+     * @param LazyLoadEvent event
+     * @memberof DotAppsConfigurationComponent
+     */
+    loadData(event?: LazyLoadEvent): void {
+        if (this.isFormContentType) {
+            this.paginationService.setExtraParams(
+                'filter',
+                this.filterInput.searchInput.nativeElement.value
+            );
+            this.paginationService
+                .getWithOffset((event && event.first) || 0)
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentType[] | DotCMSContentlet[]) => {
+                    data.forEach((item) => (item.contentType = item.variable = 'FORM'));
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
+                });
+        } else {
+            this.paginatorESService
+                .getWithOffset((event && event.first) || 0)
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentlet[]) => {
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
+                });
+        }
+    }
```

Comment:
All this logic should go into service and more specific a component store service.

In fact, we miss that the first version of this content palette should have been built using the component store.